src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -13,67 +14,162 @@
+            sec.getSectionNo()
+        )));
 
-        return null;
+        return assignmentsDTO;
     }
 
+    // TODO: trim title dto string to remove extra white spaces at start and end
     // add assignment
     // user must be instructor of the section
     // return AssignmentDTO with assignmentID generated by database
     @PostMapping("/assignments")
-    public AssignmentDTO createAssignment(
-            @RequestBody AssignmentDTO dto) {
+    public AssignmentDTO createAssignment(@RequestBody AssignmentDTO dto) {
+        Date dueDate = null;
+        try {
+            dueDate = Date.valueOf(dto.dueDate());
+        } catch (Exception e) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
 
-        // TODO remove the following line when done
+        Section sec = sectionRepository.findById(dto.secNo()).orElse(null);
 
-        return null;
+        // check incoming client dto for correct
+        // title length, section id, due date within limits, etc
+        if (sec == null) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        } else if (sec.getSecId() != dto.secId()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        } else if (dto.title().length() > 45 || dto.title().isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        } else if (!sec.getTerm().getStartDate().before(dueDate) || !sec.getTerm().getEndDate().after(dueDate)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        } else if (!sec.getCourse().getCourseId().equals(dto.courseId())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+
+        Assignment assignment = new Assignment();
+        assignment.setTitle(dto.title());
+        assignment.setDueDate(dueDate);
+        assignment.setSection(sec);
+
+        assignment = assignmentRepository.save(assignment);
+
+        return new AssignmentDTO(
+            assignment.getAssignmentId(),
+            assignment.getTitle(),
+            assignment.getDueDate().toString(),
+            assignment.getSection().getCourse().getCourseId(),
+            assignment.getSection().getSecId(),
+            assignment.getSection().getSectionNo()
+        );
     }
 
+    // TODO: trim title dto string to remove extra white spaces at start and end
     // update assignment for a section.  Only title and dueDate may be changed.
     // user must be instructor of the section
     // return updated AssignmentDTO
     @PutMapping("/assignments")
     public AssignmentDTO updateAssignment(@RequestBody AssignmentDTO dto) {
+        Date dueDate = null;
+        try {
+            dueDate = Date.valueOf(dto.dueDate());
+        } catch (Exception e) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+
+        Assignment assignment = assignmentRepository.findById(dto.id()).orElse(null);
```

Comment:
Suggestion - Could simply code with the following and additional checks would not be needed 

```Assignment assignment = assignmentRepository.findById(dto.id()).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Assignment not found"));```